nilearn/datasets/tests/test_func.py

```diff
@@ -471,75 +474,120 @@ def test_fetch_megatrawls_netmats():
+
+    # create an empty .gz file
+    f_in = open(name_f)
+    name_f_gz = os.path.join(cobre_dir,'phenotypic_data.tsv.gz')
+    f_out = gzip.open(name_f_gz, 'wb')
+    #f_out.writelines(f_in)
+    f_out.close()
+    f_in.close()
+
+    dummy = os.path.join(cobre_dir, '4197885')
     dummy_data = []
-    for i in np.hstack([ids_sch, ids_cont]):
+
+    for i in np.hstack(ids_n):
         # Func file
-        f = 'fmri_' + i + '_session1_run1.nii.gz'
-        m = 'fmri_' + i + '_session1_run1_extra.mat'
-        dummy_data.append({'downloadUrl': 'whatever', 'name': f})
-        dummy_data.append({'downloadUrl': 'whatever', 'name': m})
+        f = 'fmri_00' + str(i) + '.nii.gz'
 
-    # Add the CSV file
+        m = 'fmri_00' + str(i) + '.tsv.gz'
+        dummy_data.append({'download_url': 'whatever', 'name': f})
+        dummy_data.append({'download_url': 'whatever', 'name': m})
+
+    # Add the TSV file
+    dummy_data.append({
+        'download_url': 'whatever', 'name': 'phenotypic_data.tsv.gz'})
+    # Add JSON files
     dummy_data.append({
-        'downloadUrl': 'whatever', 'name': 'cobre_model_group.csv'})
+        'download_url': 'whatever', 'name': 'keys_confounds.json'})
+    dummy_data.append({
+        'download_url': 'whatever', 'name': 'keys_phenotypic_data.json'})
+    dummy_data.append({
+        'download_url': 'whatever', 'name': 'README.md'})
+
+    dummy_data = {'files': dummy_data}
     json.dump(dummy_data, open(dummy, 'w'))
     local_url = "file://" + dummy
 
     # All subjects
     cobre_data = func.fetch_cobre(n_subjects=None, data_dir=tst.tmpdir,
                                   url=local_url)
 
-    phenotypic_names = ['description', 'func', 'mat_files', 'phenotypic']
+    phenotypic_names = ['func', 'confounds', 'phenotypic','description',
+                           'desc_con', 'desc_phenotypic', 'readme']
+
     # test length of functional filenames to max 146
     assert_equal(len(cobre_data.func), 146)
-    # test length of corresponding matlab files of same length to max 146
-    assert_equal(len(cobre_data.mat_files), 146)
+    # test length of corresponding confounds files of same length to max 146
+    assert_equal(len(cobre_data.confounds), 146)
     # test return type variables
-    assert_equal(sorted(cobre_data), phenotypic_names)
+    assert_equal(sorted(cobre_data), sorted(phenotypic_names))
     # test functional filenames in a list
     assert_true(isinstance(cobre_data.func, list))
-    # test matlab files in a list
-    assert_true(isinstance(cobre_data.mat_files, list))
-
+    # test confounds files in a list
+    assert_true(isinstance(cobre_data.confounds, list))
     assert_true(isinstance(cobre_data.func[0], _basestring))
     # returned phenotypic data will be an array
     assert_true(isinstance(cobre_data.phenotypic, np.recarray))
     # data description should not be empty
-    assert_not_equal(cobre_data.description, '')
+    #assert_not_equal(cobre_data.description, '')
```

Comment:
We need to strive not to have commented lines of code: people in the future will wonder what they are for.

We should either create a data description for the test and uncomment this line, or give up on this test and fully remove this line.